pkg/redirectpolicy/manager.go

```diff
@@ -273,6 +274,33 @@ func (rpm *Manager) OnAddService(svcID k8s.ServiceID) {
 	}
 }
 
+// EnsureService ensures that the LRP service is updated to the latest state.
+// It is called after synchronization is complete during agent startup.
+func (rpm *Manager) EnsureService(svcID k8s.ServiceID) (bool, error) {
+	rpm.mutex.Lock()
+	defer rpm.mutex.Unlock()
+
+	if len(rpm.policyConfigs) == 0 {
+		return false, nil
+	}
+
+	if strings.HasSuffix(svcID.Name, localRedirectSvcStr) {
+		id := k8s.ServiceID{
+			Name:      strings.TrimSuffix(svcID.Name, localRedirectSvcStr),
+			Namespace: svcID.Namespace,
+		}
```

Comment:
Could use `strings.CutSuffix` here instead of `strings.HasSuffix` + `strings.CutSuffix`:

```suggestion
	if svcName, found := strings.CutSuffix(svcID.Name, localRedirectSvcStr); found {
		id := k8s.ServiceID{
			Name:      svcName,
			Namespace: svcID.Namespace,
		}
```